AdminNotificationManagerPlugin.inc.php

```diff
@@ -207,15 +224,15 @@ function _getNotificationSettingsMap() {
 	 * @return array
 	 */
 	function _getContexts() {
-		$contextDao = Application::getContextDAO();
-		$contextIterator = $contextDao->getAvailable();
-		if ($contextIterator && $contextIterator->getCount() > 1) {
-			$contextsById = array();
-			while ($context = $contextIterator->next()) {
-				$contextsById[$context->getId()] = $context->getLocalizedName();
+		$contextIdsObject = new ContextService();
+		$ids = array();
+		if ($contextIdsObject && $contextIdsObject->getCount() >= 1) {
```

Comment:
The original formulation of `if ($contextIterator && $contextIterator->getCount() > 1)` is intended to determine if the an iterator is present for the contexts, and if the iterator has anything to offer.  This is not really needed with the `ContextService`, as `getIds()` is guaranteed to return an `array`, which we know we can use in a `foreach`.

https://github.com/pkp/pkp-lib/blob/14a8480b41e1f211bc1704f5b33124ee2c3741db/classes/services/PKPContextService.inc.php#L58-L63

https://github.com/pkp/pkp-lib/blob/14a8480b41e1f211bc1704f5b33124ee2c3741db/classes/services/interfaces/EntityReadInterface.inc.php#L34-L40

It's best to remove this unneeded code for simplicity.